### PR TITLE
 NetDb: misc. datatype refactoring + documentation 

### PR DESCRIPTION
**Pull request by anonimal**
**By submitting this pull-request, I confirm the following:**
- I have read and understood the [Contributing Guide](https://github.com/monero-project/kovri/blob/master/doc/CONTRIBUTING.md).
- I have checked that another pull-request for this purpose does not exist.
- I have considered and confirmed that this submission will be valuable to others.
- I accept that this submission may not be used and that this pull-request may be closed by the will of the maintainer.
- I give this submission freely under the BSD 3-clause license.

_Place an X inside the bracket (or click the box in preview) to confirm_
- [X] I confirm.

---

Referencing https://github.com/monero-project/kovri/pull/415

FYI: you should almost always work within a topic branch, even when squashing (not in master).

### CODE REVIEW

File: src/core/net_db.cc [hunk 1]
```diff
@@ -154,34 +154,30 @@ void NetDb::Run() {
...
-            static_cast<std::uint16_t>(NetDbPrefs::MaxExploratoryTunnels);
-        } else if ((known_routers >
-            static_cast<std::uint16_t>(NetDbPrefs::MinimumKnownRouters) &&
-            known_routers <
-            static_cast<std::uint16_t>(NetDbPrefs::FavouredNumKnownRouters)) ||
-            ts - last_exploratory >=
-            static_cast<std::uint16_t>(NetDbDuration::DelayedExploreInterval)) {
-          num_routers = static_cast<std::uint16_t>(NetDbPrefs::MinExploratoryTunnels);
+        if (known_routers > 0
+            && known_routers < static_cast<std::uint16_t>(NetDbSize::MinKnownRouters)) {
```
Comment (anonimal) [inline src/core/net_db.cc:170]: We are assuming that

``` cpp
ts - last_exploratory >= static_cast<std::uint16_t>(NetDbInterval::DelayedExploratory)
```

File: src/core/net_db.cc [hunk 2]
```diff
@@ -154,34 +154,30 @@ void NetDb::Run() {
...
         Publish();
         last_publish = ts;
       }
       // builds exploratory tunnels at Nth interval to find more peers
-      if (ts - last_exploratory >=
-          static_cast<std::uint16_t>(NetDbDuration::ExploreInterval)) {
-        auto known_routers = m_RouterInfos.size();
+      if (ts - last_exploratory >= static_cast<std::uint16_t>(NetDbInterval::Exploratory)) {
+        auto known_routers = GetNumRouters();
         std::uint16_t num_routers = 0;
```
Comment (anonimal) [inline src/core/net_db.cc:164]: If I'm not mistaken, this should be initialized with `static_cast<std::uint16_t>(NetDbSize::MinExploratoryTunnels)` and then we can lose the `else if` block entirely (assuming we truly don't care about delayed exploratory time (see comment below)). I don't think we ever should have this as 0 - or else we won't explore.

File: src/core/net_db.cc [hunk 3]
```diff
@@ -154,34 +154,30 @@ void NetDb::Run() {
...
-            static_cast<std::uint16_t>(NetDbPrefs::MinimumKnownRouters) &&
-            known_routers <
-            static_cast<std::uint16_t>(NetDbPrefs::FavouredNumKnownRouters)) ||
-            ts - last_exploratory >=
-            static_cast<std::uint16_t>(NetDbDuration::DelayedExploreInterval)) {
-          num_routers = static_cast<std::uint16_t>(NetDbPrefs::MinExploratoryTunnels);
+        if (known_routers > 0
+            && known_routers < static_cast<std::uint16_t>(NetDbSize::MinKnownRouters)) {
+          num_routers = static_cast<std::uint16_t>(NetDbSize::MaxExploratoryTunnels);
+        } else if (!known_routers
```
Comment (olarks) [inline src/core/net_db.cc:172]: Why are we testing if known_routers == 0? If known_routers were zero we wouldn't want only 1 exploratory tunnel being built would we?

**Review by olarks — CHANGES_REQUESTED** [review]

File: src/core/net_db.cc (hunk 3 above)
Comment (anonimal) [reply src/core/net_db.cc:172]: This was at the end of a long day, I took the meaning out of context. I'll revert. Fortunately, I've ensured that we always have enough routers to begin (in reseed).

**Comment by anonimal** [comment]
Closing, will cherry pick into `net-db-refactor`.

**Pull request closed without merging**